srcjs/output_binding_html.js

```diff
@@ -107,23 +107,55 @@ function renderDependency(dep) {
   }
 
   if (dep.stylesheet) {
-    var stylesheets = $.map(asArray(dep.stylesheet), function(stylesheet) {
-      var this_href = href + "/" + encodeURI(stylesheet);
-      // If a styleSheet with this href is already active, then disable it
-      var re = new RegExp(this_href + "$");
-      for (var i = 0; i < document.styleSheets.length; i++) {
-        var h = document.styleSheets[i].href;
-        if (!h) continue;
-        if (restyle && h.match(re)) document.styleSheets[i].disabled = true;
-      }
-      // Force client to re-request the stylesheet if we've already seen it
-      // (without this unique param, the request might get cached)
-      if (restyle) {
-        this_href = this_href + "?restyle=" + new Date().getTime();
-      }
-      return $("<link rel='stylesheet' type='text/css'>").attr("href", this_href);
+    var links = $.map(asArray(dep.stylesheet), function(stylesheet) {
+      return $("<link rel='stylesheet' type='text/css'>").attr("href", href + "/" + encodeURI(stylesheet));
     });
-    $head.append(stylesheets);
+
+    if (!restyle) {
+      $head.append(links);
+    } else {
+      $.map(links, function(link) {
+        // Flag 'stale' stylesheets for disable+removal after new styles are inserted
+        for (var i = 0; i < document.styleSheets.length; i++) {
+          var sheet = document.styleSheets[i];
+          if (typeof sheet.href !== "string") continue;
+          // The sheet's href is a full URL
+          if (sheet.href.indexOf(link.attr("href")) === -1) continue;
+          sheet._shiny_restyle = true;
+        }
+        // Request the new stylesheet (with a timestamp to prevent caching)
+        // and insert a new inline style. Note that this could also done by inserting
+        // a the new <link> (w/ timestamp in href), then disabling the 'stale'
+        // document.styleSheets, but that approach doesn't work with IE11 (and this one does)
+        var xhr = new XMLHttpRequest();
+        var url = link.attr("href") + "?restyle=" + new Date().getTime();
+        xhr.open('GET', url);
+        xhr.onreadystatechange = function() {
```

Comment:
I think it would be better to have this xhr stuff happen only when needed (in IE), because the `link` method seems more correct.